src/WingetCreateCLI/Commands/NewLocaleCommand.cs

```diff
@@ -0,0 +1,326 @@
+                    }
+                    else
+                    {
+                        Logger.WarnLocalized(nameof(Resources.SkippingPullRequest_Message));
+                    }
+                }
+                else
+                {
+                    return false;
+                }
+
+                return await Task.FromResult(commandEvent.IsSuccessful = true);
+            }
+            catch (Octokit.NotFoundException e)
+            {
+                Logger.ErrorLocalized(nameof(Resources.Error_Prefix), e.Message);
+                Logger.ErrorLocalized(nameof(Resources.OctokitNotFound_Error));
+                return false;
+            }
+            finally
+            {
+                TelemetryManager.Log.WriteEvent(commandEvent);
+            }
+        }
+
+        private static void DisplayGeneratedLocales(List<LocaleManifest> newLocales)
+        {
+            Logger.DebugLocalized(nameof(Resources.GenerateNewLocalePreview_Message));
+            foreach (var localeManifest in newLocales)
+            {
+                Logger.InfoLocalized(nameof(Resources.LocaleManifest_Message), localeManifest.PackageLocale);
+                Console.WriteLine(localeManifest.ToYaml(true));
+            }
+        }
+
+        private static void FillLocalePropertiesForUserCompletion(LocaleManifest fromManifest, LocaleManifest toManifest, List<string> properties)
+        {
+            foreach (string propertyName in properties)
+            {
+                if (propertyName == nameof(LocaleManifest.PackageLocale))
+                {
+                    continue;
+                }
+
+                var value = fromManifest.GetType().GetProperty(propertyName).GetValue(fromManifest);
+                if (value != null)
+                {
+                    toManifest.GetType().GetProperty(propertyName).SetValue(toManifest, value);
+                }
+            }
+        }
+
+        private List<LocaleManifest> ParseArgumentsAndGenerateLocales(Manifests originalManifests)
+        {
+            LocaleManifest referenceLocaleManifest;
+
+            if (!string.IsNullOrEmpty(this.ReferenceLocale))
+            {
+                try
+                {
+                    referenceLocaleManifest = (LocaleManifest)LocaleHelper.GetMatchingLocaleManifest(this.ReferenceLocale, originalManifests);
+                }
+                catch (ArgumentException)
+                {
+                    Logger.ErrorLocalized(nameof(Resources.InvalidLocale_ErrorMessage));
+                    return null;
+                }
+
+                if (referenceLocaleManifest == null)
+                {
+                    Logger.ErrorLocalized(nameof(Resources.ReferenceLocaleNotFound_Error));
+                    return null;
+                }
+            }
+            else
+            {
+                referenceLocaleManifest = originalManifests.DefaultLocaleManifest.ToLocaleManifest();
+            }
```

Comment:
This check should have already happened before we call this method. If we change to my suggestions where we are passing in the reference locale, the logic for determining the reference locale can be removed.